prow/plugins/trigger/pull-request.go

```diff
@@ -224,12 +234,44 @@ func TrustedPullRequest(tprc trustedPullRequestClient, trigger plugins.Trigger,
 }
 
 // buildAll ensures that all builds that should run and will be required are built
-func buildAll(c Client, pr *github.PullRequest, eventGUID string, elideSkippedContexts bool) error {
+func buildAll(c Client, pr *github.PullRequest, eventGUID string, elideSkippedContexts bool, baseSHA string, presubmits []config.Presubmit) error {
 	org, repo, number, branch := pr.Base.Repo.Owner.Login, pr.Base.Repo.Name, pr.Number, pr.Base.Ref
 	changes := config.NewGitHubDeferredChangedFilesProvider(c.GitHubClient, org, repo, number)
-	toTest, toSkip, err := pjutil.FilterPresubmits(pjutil.TestAllFilter(), changes, branch, c.Config.Presubmits[pr.Base.Repo.FullName], c.Logger)
+	toTest, toSkip, err := pjutil.FilterPresubmits(pjutil.TestAllFilter(), changes, branch, presubmits, c.Logger)
 	if err != nil {
 		return err
 	}
-	return RunAndSkipJobs(c, pr, toTest, toSkip, eventGUID, elideSkippedContexts)
+	return RunAndSkipJobs(c, pr, baseSHA, toTest, toSkip, eventGUID, elideSkippedContexts)
+}
+
+// getPresubmitsAndBaseSHA is an extra closure that allows us to defer calls to GitHub when
+// inrepoconfig is disabled, without having to introduce distinct code paths in `handlePR`
+func getPresubmitsAndBaseSHA(c Client, pr github.PullRequest) (
```

Comment:
Why not DRY this out with the other one?

Reply:
Because one fetches the pr, the other one gets it passed in

Reply:
Right but the rest is identical?

Reply:
Okay, I've updated the pull, LMK which one you like more